ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.core.bnd;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.osgi.service.component.annotations.Component;
+
+import aQute.bnd.build.Project;
+
+@Component(service = IAdapterFactory.class, property = {
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=org.eclipse.core.resources.IProject",
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=aQute.bnd.build.Project" })
```

Comment:
Isn't this an unidirectional adaption from IProject to (BND)-Project?
Therefore it does not look right that `SERVICE_PROPERTY_ADAPTABLE_CLASS` is used for both source and target type.
If I understood the doc correctly it should be:
```suggestion
		IAdapterFactory.SERVICE_PROPERTY_ADAPTER_NAMES + "=aQute.bnd.build.Project" })
```

Reply:
Looking at https://github.com/bndtools/bnd/pull/5885/files#diff-798b65a162b9e5e53ee378226f1a31b3c72511f2d12c6aec6bea5c31ced1d891 seems to confirm my assumption, nevertheless I have the impression that the implementation and service properties don't really provide the opposite but the same direction. Or was it not the intend to provide an (BND-project->IProject adoption there?)

Reply:
> Or was it not the intend to provide an (BND-project->IProject adoption there?)

It should be `IProject` -> (bnd) `Project`